app/controllers/programs_controller.rb

```diff
@@ -1,4 +1,7 @@
 class ProgramsController < ApplicationController
+  before_action :require_admin
+  skip_before_action :require_admin, only: [:index, :show]
```

Comment:
I would probably scope the before action instead of having both a before and skip action

`before_action :require_admin, only: [:new, :create, :edit, :update, :destroy]`